soc/xtensa/intel_adsp/common/ipc.c

```diff
@@ -48,24 +54,31 @@ void z_intel_adsp_ipc_isr(const void *devarg)
 
 		regs->tdr = INTEL_ADSP_IPC_BUSY;
 		if (done && !IS_ENABLED(CONFIG_SOC_SERIES_INTEL_CAVS_V15)) {
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE1X
+			regs->tda = ACE_IPC_TDA1X_DONE;
+#else
 			regs->tda = INTEL_ADSP_IPC_DONE;
+#endif
 		}
 	}
 
-	/* Same signal, but on different bits in 1.5 */
-	bool done = IS_ENABLED(CONFIG_SOC_SERIES_INTEL_CAVS_V15) ?
-		(regs->idd & CAVS_IPC_IDD15_DONE) : (regs->ida & INTEL_ADSP_IPC_DONE);
+	/* Same signal, but on different bits in CAVS1.5 */
+#ifdef CONFIG_SOC_SERIES_INTEL_CAVS_V15
+	bool done = (regs->idd & CAVS_IPC_IDD15_DONE);
+#else
+	bool done = (regs->ida & INTEL_ADSP_IPC_DONE);
+#endif
 
 	if (done) {
 		if (devdata->done_notify != NULL) {
 			devdata->done_notify(dev, devdata->done_arg);
 		}
 		k_sem_give(&devdata->sem);
-		if (IS_ENABLED(CONFIG_SOC_SERIES_INTEL_CAVS_V15)) {
-			regs->idd = CAVS_IPC_IDD15_DONE;
-		} else {
-			regs->ida = INTEL_ADSP_IPC_DONE;
-		}
+#ifdef CONFIG_SOC_SERIES_INTEL_CAVS_V15
+		regs->idd = CAVS_IPC_IDD15_DONE;
+#else
+		regs->ida = INTEL_ADSP_IPC_DONE;
+#endif
```

Comment:
I'm curious about why changing `IS_ENABLED` with `ifdef` in this file.

Reply:
I think I've already answered this question.
IS_ENABLED macro expects that both CAVS_IPC_IDD15_DONE and INTEL_ADSP_IPC_DONE are defined in compilation unit.
My refactorings changes this that when you compile ACE1X it does not have CAVS_IPC_IDD15_DONE defined, because why it would have definitions for other SoC that is not currently built.
So IS_ENABLED had to be replaced with #ifdef.